Bottom: 821fa02d28ae39e1811861253577aa3867197b6f
Top:    6e1dcf855b0e9d153305fa82245ee1848bc53ca9
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-01-22 18:09:38 +0800

vhost: descriptor batch dequeuing

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/vhost.h b/drivers/vhost/vhost.h
index 79c6e7a..807828a 100644
--- a/drivers/vhost/vhost.h
+++ b/drivers/vhost/vhost.h
@@ -84,6 +84,12 @@ enum vhost_uaddr_type {
 	VHOST_NUM_ADDRS = 3,
 };
 
+struct vhost_vring_desc {
+	struct vring_desc descs[64];
+	int head;
+	int tail;
+};
+
 /* The virtqueue structure describes a queue attached to a device. */
 struct vhost_virtqueue {
 	struct vhost_dev *dev;
@@ -150,6 +156,8 @@ struct vhost_virtqueue {
 	bool user_be;
 #endif
 	u32 busyloop_timeout;
+
+	struct vhost_vring_desc descq;
 };
 
 struct vhost_msg_node {
